test/common/TestGradient.hs

```diff
@@ -288,28 +290,57 @@ testBarR =
 
 -- The following are borrowed from https://github.com/benl23x5/adops.
 
--- | Derivative of unpadded full convolution with respect to the input image.
+-- | Derivative of padded full convolution with respect to the input image.
```

Comment:
In the adops repo, we had a comment saying this is an "unpadded convolution -- output shape is the same as the input shape." The comment is wrong because if convolution doesn't pad its input, then the output should be smaller. What it should really say is "No custom padding. Only pad the input with 0s such that output is the same shape as the input." I'll update both adops and this to be clearer.

Reply:
I've fixed the docs over in adops.